### DISCUSSION TITLE
Docker with nanomq_vSomeip_gateway enabled

### DISCUSSION THREAD

**Discussion by balajirajan87** [nanomq/nanomq#1787] [Q&A]
Hello , 

I had gone through the NanoMQ Docs, but i do not understand if there is a Docker already available with **vSomeip_Gateway** enabled ? i had downloaded the 0.18.2-full docker, but in that i see that nanomq_cli is with vsomeip gateway disabled. also with that docker i tried to import the conf file (with vSOMEIP gateway specific contents set) from the host, but i see that nanomq_cli was still not enabled for vSomeIP gateway.

can anyone help me here ?

Thanks
Balaji

**Comment by JaylinYu** [comment]
Hi @balajirajan87 
The official SOME/IP Proxy (with COMMONAPI and better usability) is not available in Open-Source release, But you can build your own Docker/binary, the wire-binary proxy code are all open-source. 
BTW, we only provide ADAS/Data-closed loop func in business shipping, you can request one if there is need.

**Reply by balajirajan87** [reply]
HI @JaylinYu :

Ah ok .. understood.. Thanks !!

so i can build my own docker from the dockerfiles located: [here ](https://github.com/nanomq/nanomq/tree/master/deploy/docker) right ?

can you also please confirm the below points for a better understanding ?

1. considering i just use the dockerfile-dev, in order to enable vSOME/IP_GATEWAY (i.e, nanomq_cli vsomeip_gateway [--conf <path>]) inside the docker, do i just need to enable VSOMEIP via the cmake command: `cmake -DBUILD_VSOMEIP_GATEWAY=ON ..` or do i need to install vsomeip libraries inside the docker ?
2. I do not understand how the vsomeip Gateway within the nanomq_cli communicates with vsomeip server. is it like both the vsomeip server ("hello_world_service") and nanomq should run both inside the docker ?

Thanks a lot in advance !!
- Balaji

**Reply by JaylinYu** [reply]
1. yes, you are required to have vsomeip lib dependency ready on your own. we have to exclude the source code for license protection.
2. the someip gateway in nanomq_cli is a proxy. it only works as a bidirectional channel : SOMEIP client<->MQTT Client. Now it is an independent process which is more flexible for integration.
